### PR TITLE
apps: Add empty ip/port checks in network policies

### LINKED ISSUE #1363: [1] Network policies allow everything when ips and ports are empty
**Describe the bug**

When specifying empty list for ips  and ports for network polices it is expected to not allow that type of traffic. 
However it opens upp for all traffic.

The following example shows why:
Config file
```yaml
service:
  ips: []
  ports: []
```
Produces this network policy
```
-egress/ingress
- {}
```

Which opens up all traffic for egress or ingress.

**To Reproduce**
1. Set `ips` and `ports` to []
2. Run `kubectl get networkpolicies.networking.k8s.io -n <namespace> allow-<service> -oyaml`
3. See `{}` as a rule -> allow everything

**Expected behavior**

Not to open up all traffic

**Version (add all relevant versions):**
 - 0.27.0

**Additional context**

Need to add a check that both ips and ports are set
```
{{- if and .Values.<>.ips .Values.<>.ports }}
```

### PR DESCRIPTION
**Pull request by lunkan93**
**What this PR does / why we need it**:
Add empty ip/port checks in network policies to prevent allowing all traffic.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #1363  

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/networkpolicy/common/templates/cert-manager/startupapicheck.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ spec:
   policyTypes:
     - Egress
   egress:
```
Comment (robinAwallace) [inline helmfile/charts/networkpolicy/common/templates/cert-manager/startupapicheck.yaml:13]: Can you check if an empty egress results in `egress: {}`? If that is the case you need to move the if state

**Review by robinAwallace — COMMENTED** [review]
It looks good :+1: But check my comment

File: helmfile/charts/networkpolicy/common/templates/cert-manager/startupapicheck.yaml (hunk 1 above)
Comment (lunkan93) [reply helmfile/charts/networkpolicy/common/templates/cert-manager/startupapicheck.yaml:13]: > Can you check if an empty egress results in `egress: {}`? If that is the case you need to move the if state

Checked this and an empty egress resulted in `egress:` not being present in the resulting policy

**Review by Xartos — APPROVED** [review]
LGTM

**Review by raviranjanelastisys — APPROVED** [review]
LGTM !!

**Review by robinAwallace — APPROVED** [review]

**Pull request merged**